src/lib/y2storage/planned/partitions_distribution.rb

```diff
@@ -111,31 +117,77 @@ def partitions_count
       #
       # @return [Integer] -1, 0, 1 like <=>
       def better_than(other)
-        # The smallest gaps the better
-        res = gaps_total_size <=> other.gaps_total_size
-        return res unless res.zero?
+        criteria = [
+          # The smaller gaps the better
+          [:gaps_total_size, :smaller],
 
-        # The fewer gaps the better
-        res = gaps_count <=> other.gaps_count
-        return res unless res.zero?
+          # The fewer gaps the better
+          [:gaps_count, :smaller],
 
-        # The fewer physical volumes the better
-        res = partitions_count <=> other.partitions_count
-        return res unless res.zero?
+          # The fewer physical volumes the better
+          [:partitions_count, :smaller],
 
-        # The less fragmentation the better
-        res = spaces.count <=> other.spaces.count
-        return res unless res.zero?
+          # The bigger installation the better
+          [:spaces_total_size, :bigger],
 
-        # The biggest installation the better
-        res = other.spaces_total_size <=> spaces_total_size
-        return res unless res.zero?
+          # We want the partitions with biggest weight to be placed in the spaces
+          # in which there is more extra space to be distributed
+          [:weight_space_deviation, :smaller],
+
+          # The less fragmentation the better
+          [:spaces_count, :smaller]
+        ]
+
+        criteria.each do |criterion, order|
+          res = public_send(criterion) <=> other.public_send(criterion)
+          res *= -1 if order == :bigger
+          return res unless res.zero?
+        end
 
         # Just to ensure stable sorting between different executions in case
         # of absolute draw in all previous criteria
         comparable_string <=> other.comparable_string
       end
 
+      # A coefficient expresing how far is the distribution from the ideal
+      # situation in which there is a 1:1 correlation between the weights of the
+      # partitions in every assigned space and the extra disk in that space to
+      # be distributed.
+      #
+      # In general, for two distributions that are considered equivalent in
+      # any other aspect, the one in which the partitions with biggest weights
+      # are placed in the spaces with more "growth potential" is considered the
+      # best. This coefficient is zero if the proportion of weights (of every
+      # assigned space compared with the total) is equal to the proportion of
+      # extra spaces (same calculation).
+      #
+      # @see #better_than
+      # @return [Float] a number between 0.0 and 1.0
```

Comment:
why it return Float when better then need Integer?

Reply:
`better_than` doesn't need Integer. It needs two values that are comparable via `<=>`. For example. `#gaps_total_size` or `#spaces_total_size` are not integer either.

Reply:
ah, maybe document it on top of criteria, that it require method that return something where `<=>` can be applied

Reply:
Comment added